### PR TITLE
feature/v2.6.8

### PR DESCRIPTION
**Pull request by mattgresham**
- Added Zip purchase and refund functionality
- Added tests for Zip
- Added changes specific to SPICE

### CODE REVIEW

File: src/Purchase.js [hunk 1]
```diff
@@ -549,12 +549,12 @@ export class SignatureAccept
 
 export class MotoPurchaseRequest
 {
-    constructor(amountCents, posRefId, surchargeAmount)
+    constructor(amountCents, posRefId, surchargeAmount, SuppressMerchantPassword)
```
Comment (RaymondK-AP) [inline src/Purchase.js]: will this break backwards compatibility?

File: src/Spi.js [hunk 2]
```diff
@@ -829,6 +831,82 @@ class Spi {
...
+            this.CurrentTxFlowState.Sent(`Asked EFTPOS to refund ${refundAmount / 100}`);
+        }
+
+        document.dispatchEvent(new CustomEvent('TxFlowStateChanged', { detail: this.CurrentTxFlowState }));
+        return new InitiateTxResult(true, 'Zip Refund Initiated');
+    }
+
+    /// <summary>
+    /// Initiates a Zip purchase transaction. Be subscribed to TxFlowStateChanged event to get updates on the process.
+    /// <para>Tip and cashout are not allowed simultaneously.</para>
```
Comment (RaymondK-AP) [inline src/Spi.js]: probably don't need this line

File: tests/ZipTransactions.spec.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+import { ZipPurchaseRequest, ZipPurchaseResponse, ZipRefundRequest, ZipRefundResponse } from '../src/ZipTransactions';
+
+const purchaseAmount = 1000;
+const refundAmount = 1000;
+const posRefId = 'test';
+const encKey = '81CF9E6A14CDAF244A30B298D4CECB505C730CE352C6AF6E1DE61B3232E24D3F';
+const hmacKey = 'D35060723C9EECDB8AEA019581381CB08F64469FC61A5A04FE553EBDB5CD55B9';
+
+describe('ZipTransactions', () => {
+    it('TestZipPurchaseRequestAccountSummary', () => {
```
Comment (RaymondK-AP) [inline tests/ZipTransactions.spec.js]: Unit test naming convention is different across the libraries. Do you think we should keep it consistent?

File: src/Purchase.js (hunk 1 above)
Comment (mattgresham) [reply src/Purchase.js]: No, if a value is not passed to the constructor it still resolves as false. I changed it slightly from the original implementation to match RefundRequest - https://github.com/AssemblyPayments/spi-client-js/blob/master/src/Purchase.js#L377

File: tests/ZipTransactions.spec.js (hunk 3 above)
Comment (mattgresham) [reply tests/ZipTransactions.spec.js]: Yes we should. My bad, I missed that one.

File: src/Spi.js (hunk 2 above)
Comment (mattgresham) [reply src/Spi.js]: I've removed this line

File: tests/ZipTransactions.spec.js (hunk 3 above)
Comment (mattgresham) [reply tests/ZipTransactions.spec.js]: I've updated the names to match the naming convention

File: src/Purchase.js (hunk 1 above)
Comment (RaymondK-AP) [reply src/Purchase.js]: :+1:

File: src/Spi.js [hunk 4]
```diff
@@ -404,7 +406,7 @@ class Spi {
             // Already confirmed from Eftpos - So all good now. We're Paired also from the POS perspective.
             this._log.info("Pair Code Confirmed from POS side, and was already confirmed from Eftpos side. Pairing finalised.");
             this._onPairingSuccess();
-            this._onReadyToTransact();
+            // this._onReadyToTransact();
```
Comment (RaymondK-AP) [inline src/Spi.js]: @mattgresham We might need to be weary of this change, this means that if a pos is subscribed and acting on txflowstate changed, potentially they might be affected by this.

**Pull request merged**